go/vt/mysqlctl/backup.go

```diff
@@ -323,16 +324,23 @@ func Restore(ctx context.Context, params RestoreParams) (*BackupManifest, error)
 		return nil, ErrNoBackup
 	}
 
-	bh, err := FindBackupToRestore(ctx, params, bhs)
+	restorePath, err := FindBackupToRestore(ctx, params, bhs)
 	if err != nil {
 		return nil, err
 	}
-
+	if restorePath.IsEmpty() {
+		// This condition should not happen; but we validate for sanity
+		return nil, vterrors.Errorf(vtrpc.Code_INTERNAL, "empty restore path")
+	}
+	bh := restorePath.FullBackupHandle()
 	re, err := GetRestoreEngine(ctx, bh)
 	if err != nil {
 		return nil, vterrors.Wrap(err, "Failed to find restore engine")
 	}
-
+	params.Logger.Infof("Restore: %v", restorePath.String())
+	if params.DryRun {
+		return nil, vterrors.Errorf(vtrpc.Code_CANCELED, "Restore: dry run, aborting operation")
+	}
```

Comment:
in the dry run case, do we need to return an error?

Reply:
It's a way to break through all layers up, without anyone attempting to actually use the backup result. Do you think we should return a `nil` for error?

Reply:
my understanding is dry run is used by the user to see what steps will be taken and does all the pre-condition checks pass or not, before actually running it.

Reply:
The change was made. No longer returning an error, and now exiting gracefully. The log looks something like this:
```
restore.go:202] Restore: original tablet type=RDONLY
backup.go:289] Restore: looking for a suitable backup to restore
backup.go:340] Restore: RestorePath: [full:2022-11-28.063331.zone1-0000000102, incremental:2022-11-28.063340.zone1-0000000102, incremental:2022-11-28.063343.zone1-0000000102, incremental:2022-11-28.063348.zone1-0000000102]
restore.go:230] Restore: got a restore manifest: <nil>, err=<nil>, waitForBackupInterval=0s
restore.go:289] Dry run. No changes made
restore.go:311] Restore: changing tablet type to RDONLY
```